docs/SIL.rst

```diff
@@ -1579,6 +1593,242 @@ of functions returning uninhabited types. An ``unreachable`` instruction that
+  that rebinds the value in some manner (e.x.: apply, casts, store).
+
+* **Guaranteed**. A value with a scoped lifetime whose liveness is dependent on
+  the lifetime of some other "base" owned or guaranteed value. Consumed by
+  end_borrow instructions. The "base" value is statically guaranteed to be live
+  at all of the value's paired end_borrow instructions.
+
+* **Unowned**. A value that is only guaranteed to be instantaneously valid and
+  must be copied before the value is used in an ``@owned`` or ``@guaranteed``
+  context. This is legacy needed to model argument values with the ObjC unsafe
+  unowned argument convention. This value should never be consumed.
+
+We describe each of these semantics in more detail below.
+
+Value Ownership Kind
+~~~~~~~~~~~~~~~~~~~~
+
+Owned
+`````
+
+Owned ownership models "move only" values. We require that each such value is
+consumed exactly once along all program paths. The IR verifier will flag values
+that are not consumed along a path as a leak and any double consumes as
+use-after-frees. We model move operations via "forwarding uses" such as casts
+and transforming terminators (e.x.: `switch_enum`_, `checked_cast_br`_) that
+transform the input value, consuming it in the process, and producing a new
+transformed owned value as a result.
+
+Given any value, one can produce a new "copy" of the owned value via the usage
+of the copy_value instruction. Since in ARC a "copy" of an owned class is just a
+reference count increment, we can use owned values to break down complex ARC
+def-use graphs into individual "move only" values::
+
+  // testcase.swift.
+  func doSomething(x : Klass?) {
+    let y = x as? OtherKlass
+    switch y {
+    case .none:
+      break
+    case .some(let z):
+      use(z)
+    }
+  }
+
+  // testcase.sil. A possible SILGen lowering
+  sil [ossa] @doSomething : $@convention(thin) (@guaranteed Klass) -> () {
+  bb0(%0 : @guaranteed Klass):
+    %1 = copy_value %0 : $Klass
+    checked_cast_br %1 : $Klass to $OtherKlass, bbYes, bbNo
+
+  bbYes(%2 : @owned $OtherKlass):
+    %3 = enum $Optional<OtherKlass>, case #Optional.some!enumelt, %2 : $OtherKlass
+    br bbSucc(%3 : $Optional<OtherKlass>)
+
+  bbNo(%3 : @owned $Klass):
+    destroy_value %3 : $Klass
+    %4 = enum $Optional<OtherKlass>, case #Optional.none!enumelt
+    br bbSucc(%4 : $Optional<OtherKlass>)
+
+  bbSucc(%5 : @owned $Optional<OtherKlass>):
+    switch_enum %5 : $Optional<OtherKlass>, case #Optional.some!enumelt: bbSome, case #Optional.none!enumelt: bbNone
+
+  bbSome(%6 : @owned $OtherKlass):
+    apply %nonConsumingUseOfOtherKlass(%6) : ...
+    destroy_value %6 : $OtherKlass
+    br bbEpilog
+
+  bbNone:
+    br bbEpilog
+
+  bbEpilog:
+    %9999 = tuple()
+    return %9999 : $()
+  }
+
+Notice how our individual copy threads its way through the IR. At any single
+program point, the object is only bound to a single of our owned values.
```

Comment:
I'm not sure what "object is only bound to a single of our owned values" means. An object can be pointed to by multiple owned values. I think what this example is illustrating is the way ownership is transferred through block arguments.